### PR TITLE
Upgrade to Scala 3.1.0

### PR DESCRIPTION
**Pull request by joshuakfarrar**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -6,6 +6,6 @@ lazy val root = (project in file(".")).
     Test / test := {
       val _ = (Test / g8Test).toTask("").value
     },
-    scriptedLaunchOpts ++= List("-Xms1024m", "-Xmx1024m", "-XX:ReservedCodeCacheSize=128m", "-XX:MaxPermSize=256m", "-Xss2m", "-Dfile.encoding=UTF-8"),
```
Comment (joshuakfarrar) [inline build.sbt:9]: MaxPermSize has been removed from Java.

**Comment by joshuakfarrar** [comment]
hi, @scala-steward!

**Comment by aelfric** [comment]
Since there are some more significant differences between Scala 2.13 and Scala 3, should there maybe be a separate Scala 3 template and leave this one tracking the latest patch versions of Scala 2.13

**Comment by eed3si9n** [comment]
@aelfric There's https://github.com/scala/scala3.g8.

**Comment by SethTisue** [comment]
@joshuakfarrar seconded that for now anyway, we have separate Scala 2 and Scala 3 templates and this is the Scala 2 one. So we can't take that change, but the other changes LGTM.

However, the repo README really ought to be clear that this is the Scala 2 template, and it should link to the Scala 3 template. Can I interest you in adding that change to this PR?

**Comment by SethTisue** [comment]
> the repo README really ought to be clear that this is the Scala 2 template, and it should link to the Scala 3 template for those who want that instead

I submitted this myself as #103

**Pull request closed without merging**